core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQClientFactory.scala

```diff
@@ -551,6 +551,24 @@ private[rabbitmq] object DefaultRabbitMQClientFactory extends LazyLogging {
       }
   }
 
+  private def doTimeoutAction[A, F[_]: Effect](consumerConfig: ConsumerConfig,
+                                               timeoutsMeter: Meter,
+                                               e: TimeoutException): Task[DeliveryResult] = Task {
+    import consumerConfig._
+
+    timeoutsMeter.mark()
+
+    timeoutLogLevel match {
+      case Level.ERROR => logger.error(s"[$name] Task timed-out, applying DeliveryResult.${consumerConfig.timeoutAction}", e)
```

Comment:
It might not be true but I think I've seen some logging APIs where you can pass the level as a parameter so you don't have to do this matching here. Try to look into that.

Reply:
There are some, but not in scala-logging. I know what you mean, I saw it somewhere too, but I think it was even other framework than slf4j (I juch checked the `org.slf4j.Logger` and it doesn't have this method either).